tests/spec/README.txt

```diff
@@ -0,0 +1,6 @@
+These tests specify the expected behaviour of ehrQL, and by extension, the QM.
```

Comment:
I'm going to think aloud about `tests/spec/*` in general.

Each QL element seems to be a package, and each method seems to be in a module. However, this isn't always the case because there isn't an `AggregateFrame` QL element, for example. I think this allows us to construct a table for testing each method on each QL element. It would be good to document this structure, eventually, to make adding new QL elements easier. It would also be good to document the conventions:

* The patient table is `p`
* The ID column is `patient_id` (not simply `id`, for example)
* The event table is `e`
* The in-memory engine is accessed via a fixture
* The test itself follows the Arrange, Act, Assert pattern
* ...

I'm unsure how this structure, and, indeed, the fast backend "fits" with unit testing the query language. Do we need this structure/the fast backend (integration tests?) _and_ unit tests, after all?

Reply:
@iaindillingham and I ran into something similar -- checking that comparison operators work both ways -- which we thought lived in unit tests.